clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1213,8 +1290,19 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     // Don't do -1 here because we count on this to be the first parameter added
     // (if any).
     size_t ParamIndex = Params.size();
-    for (const ParmVarDecl *Param : InitMethod->parameters())
-      addParam(FD, Param->getType().getCanonicalType());
+    auto ParamIt = InitMethod->parameters().begin();
+    if (*ParamIt) {
+      // Add meaningful argument (not '-1') to buffer_location attribute only
+      // for an accessor pointer
+      size_t BufferLocAttrArg =
```

Comment:
Why is setting -1 here necessary?  Why doesn't omitted' just mean that?

Reply:
Also, why size_t when the handleBufferLocationProperty returns int?  or vice versa?